kafka-avro-adapter/src/cs_producer.cpp

```diff
@@ -18,140 +18,177 @@
+    avro_value_t* field, 
+    mcsapi::ColumnStoreSystemCatalogTable& table,
+    int table_idx )
+{
+    switch (avro_value_get_type(field))
+    {
+        case AVRO_STRING:
+        {
+            const char* s;
+            size_t len;
+            int r = avro_value_get_string(field, &s, &len);
+            assert(r == 0);
+            std::string str = std::string(s, len);
+            insert->setColumn(table_idx, str);
+            break;
+        }
+        case AVRO_BYTES:
+        {
+            const void* s;
+            size_t len;
+            int r = avro_value_get_bytes(field, &s, &len);
+            assert(r == 0);
+            insert->setColumn(table_idx, std::string((char*)s, len));
+
+            break;
+        }
+        case AVRO_INT32:
+        {
+            int32_t out;
+            int r = avro_value_get_int(field, &out);
+            assert(r == 0);
+            insert->setColumn(table_idx, out);
+            break;
+        }
+        case AVRO_INT64:
+        {
+            int64_t out;
+            int r = avro_value_get_long(field, &out);
+            assert(r == 0);
+            insert->setColumn(table_idx, out);
+            break;
+        }
+        case AVRO_FLOAT:
+        {
+            float out;
+            int r = avro_value_get_float(field, &out);
+            assert(r == 0);
+            insert->setColumn(table_idx, out);
+            break;
+        }
+        case AVRO_DOUBLE:
+        {
+            double out;
+            int r = avro_value_get_double(field, &out);
+            assert(r == 0);
+            insert->setColumn(table_idx, out);
+            break;
+        }
+        case AVRO_BOOLEAN:
+        {
+            int out = 0;
+            int r = avro_value_get_boolean(field, &out);
+            assert(r == 0);
+            insert->setColumn(table_idx, (bool)out);
+            break;
+        }
+        case AVRO_NULL:
+        {
+            insert->setNull(table_idx);
+            break;
+        }
+        case AVRO_ENUM:
+        {
+            int out = 0;
+            int r = avro_value_get_enum(field, &out);
+            assert(r == 0);
+            insert->setColumn(table_idx, out);
+            break;
+        }
+        case AVRO_FIXED:
+        {
+            const void* s;
+            size_t len;
+            int r = avro_value_get_fixed(field, &s, &len);
+            assert(r == 0);
+            insert->setColumn(table_idx, std::string((char*)s, len));
+            break;
+        }
+        case AVRO_MAP:
+        {
+            logger() << "Unsupported type: MAP" << std::endl;
+            break;
+        }
+        case AVRO_ARRAY:
+        {
+            logger() << "Unsupported type: ARRAY" << std::endl;
+            break;
+        }
+        case AVRO_LINK:
+        {
+            logger() << "Unsupported type: LINK" << std::endl;
+            break;
+        }
+
+        default:
+            logger() << "Unsupported type: LINK" << std::endl;
```

Comment:
Seems this comment was wrong even in the old code.
```suggestion
            logger() << "Unsupported type: " << avro_value_get_type(field) << std::endl;
```
It won't show up as a string type but it'll at least show the binary value.